libs/libc/assert/Make.defs

```diff
@@ -24,8 +24,13 @@ ifeq ($(CONFIG_STACK_CANARIES),y)
 CSRCS += lib_stackchk.c
 endif
 
+# avr32-gcc (4.4.7) doesn't understand -fno-lto.
+# Please make an exception in this case.
+
+ifeq ($(CONFIG_ARCH_CHIP_AT32UC3),n)
```

Comment:
let's check CONFIG_LTO_NONE instead

Reply:
Many thanks for your message. I have implemented your suggestion. The compilation on avr32dev1 is fine. Can you please check if it is OK?